PocketKit/Sources/Sync/CoreData+Relationships/RelationshipKeyPath.swift

```diff
@@ -0,0 +1,59 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import CoreData
+
+/// Describes a relationship key path for a Core Data entity.
+public struct RelationshipKeyPath: Hashable {
+    /// The source property name of the relationship entity we're observing.
+    let sourcePropertyName: String
+
+    let destinationEntityName: String
+
+    /// The destination property name we're observing
+    let destinationPropertyName: String
+
+    /// The inverse property name of this relationship. Can be used to get the affected object IDs.
+    let inverseRelationshipKeyPath: String
+
+    public init(keyPath: String, relationships: [String: NSRelationshipDescription]) {
```

Comment:
this method feels a bit hacky with string parsing and explicit array indexes; on the other hand, because we are using `#keypath` this should ensure that we don't accidentally have mistyped strings, so.. 😄

Reply:
Yea i added comments in the file. Its not meant to be pretty. There are probably better ways to do this. i am all for cleanup here.